### PR TITLE
🐛 adds in correct weight for amp-story-product-tag text

### LINKED ISSUE #37155: [Amp story shopping] Light text weight for product tag custom text
### Description

When product tag uses custom text rather than the default dollar value, it needs to be a lighter weight.
Screen Shot 2021-12-08 at 2 11 35 PM

### PR DESCRIPTION
**Pull request by jshamble**
fixes #37155

Before (Middle Tag) | After (Middle Tag) | 
:-------------------------:|:-------------------------:
Screen Shot 2022-01-06 at 2 16 22 PM | Screen Shot 2022-01-06 at 2 16 32 PM

### CODE REVIEW

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css [hunk 1]
```diff
@@ -13,6 +13,11 @@ amp-story-shopping-tag {
   --i-amphtml-shopping-tag-backdrop-filter: blur(20px);
 }
 
+.amp-story-shopping-product-tag-text {
+  font-family: 'Poppins', sans-serif !important;
```
Comment (gmajoulet) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:17]: Should we use a fallback system font?

Comment (jshamble) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:17]: Yes, I will Add `Arial` as the fallback font.

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css [hunk 2]
```diff
@@ -13,6 +13,17 @@ amp-story-shopping-tag {
   --i-amphtml-shopping-tag-backdrop-filter: blur(20px);
 }
 
+.amp-story-shopping-product-tag-text {
+  font-family: 'Poppins','Arial', sans-serif !important;
+  font-weight: 600 !important;
+  font-size: 11px !important;
+  line-height: 15.4px !important;
+  font-style: normal;
```
Comment (coreymasanto) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:21]: There's a trailing whitespace at the end of this line that's causing the invalid whitespace check to fail

Comment (coreymasanto) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:21]: `normal` is the default font-style value, so I think this line may be unnecessary.

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css [hunk 3]
```diff
@@ -13,6 +13,17 @@ amp-story-shopping-tag {
...
 
+.amp-story-shopping-product-tag-text {
+  font-family: 'Poppins','Arial', sans-serif !important;
+  font-weight: 600 !important;
+  font-size: 11px !important;
+  line-height: 15.4px !important;
+  font-style: normal;  
+  display: flex;
+  align-items: center;
+  letter-spacing: 0.01em;
```
Comment (coreymasanto) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:24]: Add `!important` to these properties so that they can't be overridden

Comment (coreymasanto) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:24]: Does this letter-spacing value make a necessary visual difference? I'm reading [here](https://developer.mozilla.org/en-US/docs/Web/CSS/letter-spacing#internationalization_concerns) that some written languages (such as Arabic) shouldn't have any letter spacing applied, so I'm wondering if publishers using such languages would encounter issues if we specify the spacing.

@gmajoulet have any thoughts here?

Comment (jshamble) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:23]: Good point, I have removed the line spacing property.

**Review by gmajoulet — APPROVED** [review]

**Review by gmajoulet — DISMISSED** [review]

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js [hunk 4]
```diff
@@ -38,7 +38,12 @@ const renderShoppingTagTemplate = (tagData) => (
         }
       ></span>
       <span class="amp-story-shopping-tag-pill-text">
-        {tagData['product-tag-text'] || '$' + tagData['product-price']}
+        {(tagData['product-tag-text'] && (
+          <span class="amp-story-shopping-product-tag-text">
+            {tagData['product-tag-text']}
+          </span>
+        )) ||
+          '$' + tagData['product-price']}
```
[the remark this answers is not in the document]
Comment (jshamble) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js]: Yes, I have added it, however I have taken out the parameters for now as we want the dollar sign small for now. Will try to find a way to integrate this once we discuss currency with the desingers.

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css [hunk 5]
```diff
@@ -13,6 +13,15 @@ amp-story-shopping-tag {
   --i-amphtml-shopping-tag-backdrop-filter: blur(20px);
 }
 
+.amp-story-shopping-product-tag-text {
+  font-family: 'Poppins','Arial', sans-serif !important;
```
Comment (coreymasanto) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:17]: I don't think it's necessary to specify `font-family` here since it's already specified with the exact same value in the `amp-story-shopping-tag` rule above, so anything within the shopping tag (e.g., elements with the `"amp-story-shopping-product-tag-text"` class) should inherit the rule's `font-family`

**Review by coreymasanto — APPROVED** [review]

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css (hunk 5 above)
Comment (jshamble) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css]: Yep, you're right, good call.

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js [hunk 6]
```diff
@@ -38,7 +38,18 @@ const renderShoppingTagTemplate = (tagData) => (
...
+            {tagData['product-tag-text']}
+          </span>
+        )) ||
+          new Intl.NumberFormat('en-EN', {
+            style: 'currency',
+            currency: 'USD',
+          }).formatToParts(0.0)[0].value +
+            Number(
+              new Intl.NumberFormat('en-EN').format(tagData['product-price'])
+            )}
```
Comment (gmajoulet) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js:52]: Does this work for:
- Currencies that have their symbol after the number ($1,000 vs 1.000€)
- Currencies that have a space between the symbol and the number
?

Also, both the language and currency need to be dynamic.

Comment (processprocess) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js]: This work will be part of #37246

File: extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css [hunk 7]
```diff
@@ -1,5 +1,5 @@
 amp-story-shopping-tag {
-  font-family: 'Poppins', sans-serif !important;
+  font-family: 'Poppins','Arial', sans-serif !important;
```
Comment (processprocess) [inline extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:2]: Please revert this line and default to `sans-serif` to follow the [font stack](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.css#L11) pattern of our codebase.

Updating the font stack might be a good idea but that would be a global change for a separate PR.

Comment (jshamble) [reply extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css:2]: 'sans-serif' is just a font modifier right, not an actual font.
So I'm not sure what it would fall back to If I don't add in something else like Arial here.

Was adding it per Gabriel's request here:
https://github.com/ampproject/amphtml/pull/37188#discussion_r768158694

Anyway, I removed it - we can take care of it in another PR if we need it.
